src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((grouped: GroupsMap<T>, item: T) => {
+    const keyV = item[key] as string;
```

Comment:
```suggestion
    const keyV = String(item[key]);
```
Since theoretically value of the object may be anything it's better to transform it into string instead of using type casting - FYI